vpr/src/place/net_cost_handler.cpp

```diff
@@ -1757,3 +1755,108 @@ std::pair<t_bb, t_bb> NetCostHandler::union_2d_bb_incr(ClusterNetId net_id) cons
 
     return std::make_pair(merged_num_edge, merged_bb);
 }
+
+std::pair<vtr::NdMatrix<double, 3>, vtr::NdMatrix<double, 3>> NetCostHandler::estimate_routing_chann_util() const {
+    const auto& cluster_ctx = g_vpr_ctx.clustering();
+    const auto& device_ctx = g_vpr_ctx.device();
+
+    auto chanx_util = vtr::NdMatrix<double, 3>({{(size_t)device_ctx.grid.get_num_layers(),
+                                                 device_ctx.grid.width(),
+                                                 device_ctx.grid.height()}},
+                                               0);
+
+    auto chany_util = vtr::NdMatrix<double, 3>({{(size_t)device_ctx.grid.get_num_layers(),
+                                                 device_ctx.grid.width(),
+                                                 device_ctx.grid.height()}},
+                                               0);
+
```

Comment:
Give some idea of what the algorithm is.  You are assuming the expected wiring in each dimension is evenly spread over the net's bounding box. You can also comment on how / why the expected wirelength is allocated to different dimensions.